### PR TITLE
Regions Contacted on Multi Region Diagnostics Fix

### PR DESCRIPTION
**Pull request by tvaron3**
# Description
The `regionsContacted` field in the diagnostics was displaying the wrong region for multi-region accounts that don't specify a preferred regions. We were returning the first location in a map of the location endpoints and the regions, so not necessarily the location related to the default endpoint. Now it should return the first region in the available write locations list. https://github.com/Azure/azure-sdk-for-java/issues/40696

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #### Breaking Changes
 
 #### Bugs Fixed
+* Fixed an issue where `contactedRegions` shows the wrong region in a multi region account if no preferred regions are specified. - See [PR 41107]()
```
Comment (tvaron3) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:10]:
```suggestion
* Fixed an issue where `contactedRegions` shows the wrong region in a multi region account if no preferred regions are specified. - See [PR 41045](https://github.com/Azure/azure-sdk-for-java/pull/41045/files)
```

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@
 #### Breaking Changes
 
 #### Bugs Fixed
+* Fixed an issue where `contactedRegions` shows the wrong region in a multi region account if no preferred regions are specified. - See [PR 41045](https://github.com/Azure/azure-sdk-for-java/pull/41045/files)
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:10]:
```suggestion
* Fixed an issue where `contactedRegions` shows the wrong region in a multi region account if no preferred regions are specified. - See [PR 41045](https://github.com/Azure/azure-sdk-for-java/pull/41045)
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/LocationCache.java [hunk 3]
```diff
@@ -422,7 +422,7 @@ public String getRegionName(URI locationEndpoint, com.azure.cosmos.implementatio
         }
 
         //If preferred list is not set, locationEndpoint will be default endpoint, so return the hub region
-        return this.locationInfo.availableWriteEndpointByLocation.keySet().iterator().next();
+        return this.locationInfo.availableWriteLocations.get(0);
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/LocationCache.java:425]: What's the difference between `availableWriteEndpointByLocation` and `availableWriteLocations`?
I am curious if `availableWriteEndpointByLocation` is getting updated with any change in the endpoints (by adding region / removing region), and if `availableWriteLocations` follows the same behavior?
This is also used in Consistency session improvements that @jeet1995 worked on, lets connect and make sure the `getRegionName()` is returning the correct region name with this change across different features.

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/LocationCache.java:425]: I think - Tomas correct me if I misread the intention - the problem is that the keySet is not an ordered list - while availableWriteLocations is.

Comment (tvaron3) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/LocationCache.java:425]: Fabian that is correct. Both get updated at the same time. In each update, `availableWriteEndpointByLocation` is created from the new `availableWriteLocations`. The difference is that `availableWriteEndpointByLocation` is a map so the ordering gets messed up while `availableWriteLocations` is a list.

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - thanks

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/LocationCache.java (hunk 3 above)
Comment (jeet1995) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/LocationCache.java:425]: From the session consistency perspective - when we pass empty `preferredRegions` we are pinned to the primary region - so the first available region returned by the gateway. One weird scenario that could possibly happen is if the primary somehow changes prior to routing and once response is received both from diagnostic and session tracking perspective. cc: @tvaron3, @kushagraThapar, @FabianMeiswinkel

**Pull request merged**